src/ServiceManagerBase.sol

```diff
@@ -136,24 +152,24 @@ abstract contract ServiceManagerBase is OwnableUpgradeable, ServiceManagerBaseSt
      * The ServiceManager owner can eject any operators that have yet to completely migrate fully to operator sets.
      * This final step of the migration process will ensure the full migration of all operators to operator sets.
      * @param operators The list of operators to eject for the given OperatorSet
-     * @param operatorSet The OperatorSet to eject the operators from
+     * @param operatorSetId This AVS's operatorSetId to eject operators from
      * @dev The RegistryCoordinator MUST set this ServiceManager contract to be the ejector address for this call to succeed
      */
     function ejectNonmigratedOperators(
         address[] calldata operators,
-        IOperatorSetManager.OperatorSet calldata operatorSet
+        uint32 operatorSetId
     ) external onlyOwner {
         require(
-            operatorSet.avs == address(this),
-            "ServiceManagerBase.ejectNonmigratedOperators: operatorSet is not for this AVS"
-        );
-        require(
-            operatorSet.id < _registryCoordinator.quorumCount(),
+            operatorSetId < _registryCoordinator.quorumCount(),
             "ServiceManagerBase.ejectNonmigratedOperators: operatorSet does not exist"
         );
+        IOperatorSetManager.OperatorSet memory operatorSet =
+            IOperatorSetManager.OperatorSet({avs: address(this), id: operatorSetId});
         for (uint256 i = 0; i < operators.length; ++i) {
             require(
-                !_operatorSetManager.isOperatorInOperatorSet(operators[i], operatorSet),
+                _operatorSetManager.avsOperatorStatus(address(this), operator)
+                    == IOperatorSetManager.OperatorAVSRegistrationStatus.REGISTERED
+                    && !_operatorSetManager.isOperatorInOperatorSet(operators[i], operatorSet),
```

Comment:
we should not boot them out of all quorums they're in? only the one corresponding to `operatorSetId`

Reply:
https://github.com/Layr-Labs/eigenlayer-middleware/pull/263/commits/5bd70a6fea33045c1ab98b79265ec1ff42c79577